pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -93,21 +93,19 @@ type Propagator struct {
 	gc        *GarbageCollector
 }
 
-// insertNode insert the node to p.uidToNode; then it finds all owners as listed
-// in n.owners, and adds the node to their dependents list. ownerExists
-// indicates if any owner listed in n.owners exists in p.uidToNode.
-func (p *Propagator) insertNode(n *node) (ownerExists bool) {
-	p.uidToNode[n.identity.UID] = n
-	for _, owner := range n.owners {
+// insertToOwners adds n to owners' dependents list. If the owner does not exist
+// in the p.uidToNode yet, a "virtual" node will be created to represent the
+// owner. The "virtual" node will be enqueued to the dirtyQueue, so that
+// processItem() will verify if the owner exists according to the API server.
+func (p *Propagator) insertToOwners(n *node, owners []metatypes.OwnerReference) {
```

Comment:
Changed to "addDependentToOwners". Also renamed the remove function to "removeDependentFromOwners".